### ISSUE TITLE
Is this support for train_on_epoch() method?

### ISSUE THREAD

**Issue by toculin38** [run-ai/runai#13]
Since I am training for GAN, do I just assign STEPS for the optimizer then call train_on_batch? It will be very nice if you can provide some train_on_epoch usage examples.

**Comment by levosos** [comment]
Hi, and thank you for reaching out!

As explained [here](https://www.pyimagesearch.com/2018/12/24/how-to-use-keras-fit-and-fit_generator-a-hands-on-tutorial/):

> The `train_on_batch()` function accepts a single batch of data, performs backpropagation, and then updates the model parameters.

In oppose to `fit()`, the method `train_on_batch()` does not receive the batch size as an argument, but receives a numpy array of training data as an input `x` (specified in the [documentation](https://keras.io/models/model/#train_on_batch)).

The argument `x`, is actually an entire batch of samples. The argument `y` should also be of the same size.

Therefore, I'd say that you need to specify `STEPS` to the Gradient-Accumulated optimizer, just like in the examples, but make sure to pass the **mini-batch** to `train_on_batch()`, and to execute `train_on_batch()` the correct number of times.

Let's say you want to use Gradient Accumulation in order to train your model with a batch size of 128 across 4 steps, and let's say you have 128,000 samples in your dataset, so w/o GA it would have taken you 1000 steps to iterate. So you need to:

1. Pass `STEPS=4` to the Gradient-Accumulated optimizer
2. Pass a batch size of 128/4=32 (global batch size / # of steps) to each call of `train_on_batch()`
3. Iterate and call `train_on_batch()` (128,000/128)*4=4000 times (the size of the data set, divided by the global batch size, multiplied by the number of steps - **or** - 128,000/32=4000 - the size of the dataset, divided by the mini-batch size).

Unfortunately, we couldn't find any official GAN example for Keras that I could use as an example for Gradient Accumulation, but we don't see any reason why this should not work, and we are pretty certain that you'll be able to add GA support to your GAN model.

If you are able to share your model - or better, a minimal working version of it, using official datasets and methods - we would gladly add this as an official example to the library using GA.

Please let us know if you're having some trouble with using the library in your GAN model and feel free to add code samples here so we could elaborate on every step, and thanks again for reaching out!
